test/com/timezynk/mongo/test/fetch_and_replace.clj

```diff
@@ -0,0 +1,64 @@
+(ns com.timezynk.mongo.test.fetch-and-replace
+  (:require
+   [clojure.test :refer [deftest is testing use-fixtures]]
+   [com.timezynk.mongo :as m]
+   [com.timezynk.mongo.test.utils.db-utils :as dbu]))
+
+(use-fixtures :once #'dbu/test-suite-db-fixture)
+(use-fixtures :each #'dbu/test-case-db-fixture)
+
+(deftest empty-replace
+  (testing "Update with empty list"
+    (let [res (m/fetch-and-replace-one! :coll
+                                        {}
+                                        [])]
+      (is (nil? res)))
+    (let [res (m/fetch-and-replace-one! :coll
+                                        {}
+                                        '())]
+      (is (nil? res)))))
+
+(deftest bad-replace
+  (is (thrown-with-msg? IllegalArgumentException
+                        #"replacement can not be null"
+                        (m/fetch-and-replace-one! :coll
+                                                  {}
+                                                  nil))))
+
+(deftest replace-and-fetch-old
+  (m/insert! :coll {:name "1"})
+  (is (= {:name "1"}
+         (-> (m/fetch-and-replace-one! :coll
+                                       {}
+                                       {:mail "2"})
+             (dissoc :_id))))
+  (is (= {:mail "2"}
+         (-> (m/fetch-one :coll)
+             (dissoc :_id))))
```

Comment:
This asserts two things:
1. the document was indeed replaced
1. no other document is present in the collection

The latter looks out of place - unintended perhaps?

Reply:
If only comment here, but consider the same to apply to the rest of the tests in this PR.